async.go

```diff
@@ -0,0 +1,175 @@
+package plc
+
+import (
+	"sync"
+)
+
+type asyncer interface {
+	Wait() error
+	Add(name string, value interface{})
+	AddError(err error)
+}
+
+type action func(string, interface{}) error
+
+type job struct {
+	name  string
+	value interface{}
+}
+
+var noJob = job{}
+
+type async struct {
+	action
+	jobs        chan job
+	cancel      chan struct{}
+	readyToWait chan struct{}
+	wg          sync.WaitGroup
+	errOnce     sync.Once
+	err         error
+}
+
+func newAsync(act action) *async {
+	as := &async{
+		action:      act,
+		jobs:        make(chan job, 1),
+		cancel:      make(chan struct{}),
+		readyToWait: make(chan struct{}),
+	}
+
+	as.launchRoutine(noJob)
```

Comment:
I don't think there's a big loss to having the loop below spawn the first routine. And what if there are no jobs? Then that's faster :P
If your were to draw this out as a block diagram, this special case just raises lots of questions. And if you wanted to change behaviour a little (ie. add a "launchJobMonitor" to each routine, and launchroutine returns a handle, etc.) it has to be changed in two places.